apps/voting/contracts/Voting.sol

```diff
@@ -213,6 +240,70 @@ contract Voting is IForwarder, AragonApp {
         _executeVote(_voteId);
     }
 
+    /**
+    * @notice Assign `_delegate` as the delegate for the sender
+    * @param _delegate address to delegate to
+    */
+    function setDelegate(address _delegate) external {
+        require(_delegate != address(0), ERROR_ZERO_ADDRESS_PASSED);
+        require(_delegate != msg.sender, ERROR_SELF_DELEGATE);
+
+        address prevDelegate = delegates[msg.sender].delegate;
+        require(_delegate != prevDelegate, ERROR_DELEGATE_SAME_AS_PREV);
+
+        if (prevDelegate != address(0)) {
+            _removeDelegatedAddressFor(prevDelegate, msg.sender);
+        }
+        _addDelegatedAddressFor(_delegate, msg.sender);
+    }
+
+    /**
+    * @notice Unassign `_delegate` from the sender
+    */
+    function resetDelegate() external {
+        address prevDelegate = delegates[msg.sender].delegate;
+        require(prevDelegate != address(0), ERROR_DELEGATE_NOT_SET);
+
+        _removeDelegatedAddressFor(prevDelegate, msg.sender);
+        delete delegates[msg.sender];
```

Comment:
clearing of delegates mapping happening after the event `ResetDelegate` has been emited which is a bit confusing

Reply:
Thanks for pointing this out. As mentioned earlier, I'll do some refactoring to ensure that `ResetDelegate` will be emitted after the deletion.